llvm/tools/file-table-tform/file-table-tform.cpp

```diff
@@ -289,6 +367,19 @@ int main(int argc, char **argv) {
     CHECK_AND_EXIT(makeUserError("no inputs"));
   std::string &InputFile = *CurInput++;
 
+  // ensure that if copy_single_file is specified, it must be the last tform
+  bool HasCopySingleFileTform = false;
+  for (auto &P : Cmds) {
+    if (HasCopySingleFileTform) {
+      CHECK_AND_EXIT(
+          makeUserError("copy_single_file must be the last transformation"));
+    }
+    if (P.second->Kind != OPT_COPY_SINGLE_FILE) {
+      continue;
+    }
+    HasCopySingleFileTform = true;
```

Comment:
I see your change suggestion and raise you the even conciser
```
HasCopySingleFileTform = P.second->Kind == OPT_COPY_SINGLE_FILE;
```
which is sufficient given that `HasCopySingleFileTform` must be `false` when this line is reached (because of the check above). However, I am inclined to think that your suggestion preserves a bit more information so if you prefer it to this one-line alternative I am not opposed to sacrificing a bit of brevity to make intentions more clear. 😄